internal/querycoordv2/meta/resource_manager.go

```diff
@@ -671,6 +742,7 @@ func (rm *ResourceManager) transferOneNodeFromRGToRG(sourceRG *ResourceGroup, ta
 	if sourceRG.NodeNum() == 0 {
 		return -1, ErrNodeNotEnough
 	}
+
 	// TODO: select node by some load strategy, such as segment loaded.
 	node := sourceRG.GetNodes()[0]
```

Comment:
It seems that lost a node selection policy with `node filter` at `transferOneNodeFromRGToRG`.
Here we are always transfer the first node into target, but not the node that pass the `node filter`.